SimTracker/SiPixelDigitizer/plugins/SiPixelChargeReweightingAlgorithm.cc

```diff
@@ -266,12 +273,11 @@ bool SiPixelChargeReweightingAlgorithm::hitSignalReweight(const PSimHit& hit,
     printCluster(pixrewgt);
   }
 
-  for (int row = 0; row < TXSIZE; ++row) {
-    for (int col = 0; col < TYSIZE; ++col) {
+  for (int row = rowmin; row < rowmax; ++row) {
+    for (int col = colmin; col < colmax; ++col) {
       float charge = 0;
```

Comment:
dead assignment (compiler will optimize, I suppose;

Reply:
In lines from 216 to 228 rowmin, rowmax, colmin, colmax are not taken into account, the loop 216-220 seems to be an overhead.

Reply:
Good point, thanks. This is now fixed.

Reply:
@civanch, in L216-220 the full range is needed to ensure the entire `pixrewgt` array is initialized to 0. Similarly for L222-228, the full range ensures that all element of the vectors `xdouble` and `ydouble` are correctly assigned and I checked in https://github.com/cms-sw/cmssw/blob/CMSSW_12_1_0_pre3/Geometry/TrackerGeometryBuilder/interface/RectangularPixelTopology.h#L116-L129 that going beyond physical bounds of the pixel module does not present a problem for the `isItBigPixelInX(const int ixbin)` and `isItBigPixelInY(const int iybin)` methods (elements that are out of physical bounds I would anyways have to set to `false` so I get this for "free" in the same loop).

Reply:
to be consistent I would fill xdouble and ydouble by zeros (or other default) and after define values in the valid intervals.

Reply:
@civanch, I simply wanted to avoid introducing unnecessary loops since the last two commits went in the direction of optimizing the code. Keep in mind that here we are dealing with rare instances of pixel clusters on module edges. So probably 99% of the time clusters are fully inside modules and we get loops with full ranges anyways so with your proposal we would have the first full loop to set all vector elements to `false` and then the second full loop to set them to `false` again because big pixels are only on the readout chip edges so only in relatively rare instances an element will be set to `true`. Perhaps leaving a comment in the code would be a good compromise (the code would be more readable and there would be no unnecessary loops).